src/server.ts

```diff
@@ -90,31 +99,59 @@ function throwResponseValidationError(tag: string, originalValue: any, e: oar.Ma
   return null as any;
 }
 
-function voidify(value: {} | undefined) {
-  if (value == null || Object.keys(value).length > 0) {
+function lowercaseObject(o?: {} | undefined | null) {
+  if (o == null) {
+    return o;
+  }
+  const result: any = {};
+  for (const key of Object.keys(o)) {
+    result[key.toLowerCase()] = (o as any)[key];
+  }
+  return result;
+}
+
+function voidify(value: {} | undefined | null) {
+  if (value && Object.keys(value).length > 0) {
     return value;
   }
   return null;
 }
 
+function cleanHeaders<H>(maker: oar.Maker<any, H>, headers: {} | null | undefined) {
+  const normalized = voidify(lowercaseObject(headers));
+  const made = maker(normalized, {
```

Comment:
when maker is Make<any, null> this fails as it does not know what to do with the headers it gets